src/modules/name-manager/PermanentNameModule.js

```diff
@@ -0,0 +1,376 @@
+    }
+
+    return new Promise((resolve, reject) => {
+      if (this.deedOwnerVal !== super.addressVal) {
+        return reject({
+          error: 'Redeeming address provided is not the owner!'
+        });
+      }
+      const data = this.oldDeedContract.methods
+        .releaseDeed(super.labelHashVal)
+        .encodeABI();
+      const obj = {
+        from: super.addressVal,
+        to: OLD_ENS_ADDRESS,
+        data: data,
+        value: 0
+      };
+
+      super.web3Val.eth
+        .sendTransaction(obj)
+        .then(() => {
+          resolve({ success: 'Deed released succesfully!' });
+        })
+        .catch(reject);
+    });
+  }
+
+  setIPFS(file) {
+    if (super.ownerVal === '0x') {
+      throw new Error('Owner not set! Please initialize module properly!');
+    }
+
+    if (super.networkVal.type.name !== 'ETH') {
+      throw new Error('Ipfs not supported in this network!');
+    }
+
+    return new Promise((resolve, reject) => {
+      try {
+        uploadFileToIpfs(file)
+          .then(getHashFromFile)
+          .then(hash => {
+            const ipfsToHash = `0x${contentHash.fromIpfs(hash)}`;
+            const tx = {
+              from: super.addressVal,
+              to: super.resolverAddressVal,
+              data: super.resolverContractVal.methods
+                .setContentHash(super.nameHashVal, ipfsToHash)
+                .encodeABI(),
+              value: 0
+            };
+
+            super.web3Val.eth.sendTransaction(tx).then(() => {
+              return resolve({
+                success:
+                  'Transaction sent! Please wait a couple minutes to confirm changes'
+              });
+            });
+          });
+      } catch (e) {
+        reject(e);
+      }
+    });
+  }
+
+  // DNS claim name method
+  claim() {
+    return new Promise((resolve, reject) => {
+      try {
+        this.dnsClaimVal
+          .submit({
+            from: this.account.address
+          })
+          .then(resolve);
+      } catch (e) {
+        reject(e);
+      }
+    });
+  }
+
+  async _initModule() {
+    // initial valaue for the variables
+    const formValues = {
+      deedValue: 'deedValueVal',
+      deedOwner: 'deedOwnerVal',
+      secretPhrase: 'secretPhraseVal',
+      expiration: 'expirationVal',
+      expired: 'expiredVal',
+      redeemable: 'redeemableVal',
+      // Contracts
+      oldEnsContract: 'oldEnsContractVal',
+      oldDeedContract: 'oldDeedContractVal',
+      dnsRegistrarContract: 'dnsRegistrarContractVal',
+      dnsClaim: 'dnsClaimVal',
+      dnsStatus: 'dnsStatusVal'
+    };
+
+    Object.keys(formValues).forEach(propName => {
+      Object.defineProperties(this, propName, {
+        enumerable: true,
+        get: () => {
+          return this[formValues[propName]];
+        },
+        set: value => {
+          this[formValues[propName]] = value;
+        }
+      });
+    });
+    try {
+      await super._initializeNameModuleVal
```

Comment:
niiiice

Reply:
id put all these inside Promise.all([this._setDeeds,...]) this way everything will happen in parallel